hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -107,7 +107,13 @@ public void initialize(String inputName, Map<String, String> properties) {
             ? new HadoopFileIO(conf)
             : CatalogUtil.loadFileIO(fileIOImpl, properties, conf);
 
-    this.clients = new CachedClientPool(conf, properties);
+    if (properties.containsKey(CatalogProperties.CLIENT_POOL_IMPL)) {
```

Comment:
Nit: not that I have an opinion which style is better, but  can we make this block more closely follow the previous block in terms of style (using the question-mark colon operator).  So code is more consistent to understand at a glance.